app/src/main/java/com/movies/presentation/view/search_and_filter/SearchAndFilterView.kt

```diff
@@ -72,18 +71,16 @@ class SearchAndFilterView @JvmOverloads constructor(
         }
     }
 
-    fun categoryButtonListener(callback: (Category) -> Unit) {
-        categoryAdapter.onItemClickListener {
-            callback(it)
-            CategoryList.editActiveCategories(it)
-            categoryAdapter.submitList(CategoryList.getCategories())
-            categoryAdapter.notifyDataSetChanged()
+    fun categoryButtonListener(callBack: (CategoryType) -> Unit) {
+        categoryAdapter = CategoryAdapter {
+            callBack(it)
         }
+        initRecycler()
     }
 
     private fun initRecycler() {
         binding.categoryRecyclerView.adapter = categoryAdapter
```

Comment:
you set adapter to the categoryRecyclerView everytime category changes, please improve logic to set it only once

Reply:
<img width="767" alt="image" src="https://github.com/DavitTsiskarashvili/Movies/assets/99031977/0d0f056a-3321-40ee-b181-91487ff605a7">

I call initRecycler only once in categoryButtonListener() which also is called only once from fragment.